### ISSUE TITLE
Booking option duration field no longer available

### ISSUE THREAD

**Issue by monsieurmonsieur** [Wunderbyte-GmbH/moodle-mod_booking#750]
Hi, 

Maybe not really a bug, but we upgraded to mod_booking 8.2.4 and discovered the 'duration' field is no longer present when going to: Edit booking option.

In the table mdl_booking_options, the column 'duration' still exists and still contains the previous values edited for existing booking options, but it is no longer incremented when creating a new booking option with booking 8.2.4.

We are currently testing the PRO version as we thought the duration of booking options was now handled this way but couldn't find anything.

Is there still a way to type a duration manually in version 8 ?

Thanks in advance and best regards

**Comment by monsieurmonsieur** [comment]
Hello,
could it be possible to have an update on this issue ?

Thank you and best regards

**Comment by dasistwas** [comment]
@bernhard-wunderbyte could you investigate what happened to the duration field and why it is present in the table but not in the settings. Could it be that it has to be activated there:
/admin/settings.php?section=modsettingbooking

selflearningcourseactive

If it has been discontinued we also should remove the column in the table. 

@monsieurmonsieur as an alternative maybe you can use the optiondates? I think that's what replaces the duration field. Or you can make use of custom option fields. These can be used to display additional info for booking options.

**Comment by bernhard-wunderbyte** [comment]
@dasistwas Yes, the "duration" field is now used for our new "self-learning courses" feature. With self-learning courses, you can create Booking options linked to a Moodle course (online course) that has no specific start time but is starting whenever a user books. From this moment in time on, the duration will be counted. So a user could have access to the Moodle course for the defined duration (e.g. 1 month or 1 semester).

For normal booking options, you should use the default "Dates" feature or if you really want to enter a duration without using the self-learning course functionality, you could of course just define a custom booking option field called "myduration" (or something similar) like @dasistwas suggested.

**Comment by monsieurmonsieur** [comment]
Thank you both for your replies ! I will look into the different options.
